### PR TITLE
Update flux to latest release

### PR DESCRIPTION
**Pull request by paurosello**
- Update to latest release 0.15.0
- Use EmptyDir instead of PVC
- Allow use of other providers other than Github
- Move image versions to values

### CODE REVIEW

**Comment by ubergesundheit** [comment]
Seems reasonable. Some remarks

- Please add changelog entries
- While I understand its not entirely up to best practice, I specifically left the image versions as is in the final helm templates to have always the same versions as in upstream kustomization
- If you want to go forward with having image version tags configurable, I would strongly suggest you change the [`kustomization.yaml`](https://github.com/giantswarm/flux-app/blob/master/hack/kustomization.yaml) as well
- Have used the kustomization in `hack` directory?

**Comment by paurosello** [comment]
> Seems reasonable. Some remarks
> 
> * Please add changelog entries
Ups yeah, will do.

> * While I understand its not entirely up to best practice, I specifically left the image versions as is in the final helm templates to have always the same versions as in upstream kustomization
I propose this change as there are no additional changes on the upstream install file other than image versions and CRDs.

> * If you want to go forward with having image version tags configurable, I would strongly suggest you change the [`kustomization.yaml`](https://github.com/giantswarm/flux-app/blob/master/hack/kustomization.yaml) as well
Will check it out

> * Have used the kustomization in `hack` directory?
No, but I will try it out now and see if it works

**Comment by paurosello** [comment]
Ok, I did the changes using the hacks and kustomize now. I updated everything manually because I didn't read the README :)

**Comment by paurosello** [comment]
@ubergesundheit let me know if this is good and I would make a release

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+- Bump upstream flux toolkit version to 0.15.0
```
Comment (ubergesundheit) [inline CHANGELOG.md:10]:
```suggestion
- Bump upstream flux toolkit version to [0.15.0](https://github.com/fluxcd/flux2/releases/tag/v0.15.0)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Use kubectl kustomize to update app 
+- Use EmptyDir instead of PVC
+- Allow use of other providers other than Github
```
Comment (webwurst) [inline CHANGELOG.md:13]:
```suggestion
- Allow use of other providers than Github
```

**Review by ubergesundheit — APPROVED** [review]
LGTM if tested :)

**Comment by paurosello** [comment]
yup, tested and will do additional testing with customer this week

**Pull request merged**